torch/distributed/rpc/backend_registry.py

```diff
@@ -1,24 +1,34 @@
 from __future__ import absolute_import, division, print_function, unicode_literals
 
 import collections
+import datetime
 import enum
 
 import torch.distributed as dist
 import torch.distributed.distributed_c10d as dc10d
 
+from . import constants as rpc_constants
 
-BackendValue = collections.namedtuple("BackendValue", ["init_backend_handler"])
+
+BackendValue = collections.namedtuple(
+    "BackendValue", ["construct_rpc_agent_options_handler", "init_backend_handler"]
+)
 
 # Create an enum type, `BackendType`, with empty members.
 BackendType = enum.Enum(value="BackendType", names={})
 
 
-def register_backend(backend_name, init_backend_handler):
+def register_backend(
+    backend_name, construct_rpc_agent_options_handler, init_backend_handler
+):
     """Registers a new RPC backend.
 
     Arguments:
-        backend (str): backend string to identify the handler.
-        handler (function): Handler that is invoked when the
+        backend_name (str): backend string to identify the handler.
+        construct_rpc_agent_options_handler (function):
```

Comment:
this is a pretty long name, but I don't have a better suggestion. Does `rpc_agent_options_factory` work? If we do that `init_backend_handler` will become `backend_factory`.

Reply:
I don't think it is a backend_factory, it's creating the same type of backend. I can't think of a better name.